platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/ProviderIntegrationTest.groovy

```diff
@@ -274,6 +274,27 @@ class ProviderIntegrationTest extends AbstractIntegrationSpec {
                 - task ':run' property 'p2'""".stripIndent())
     }
 
+    def "reasonable error message if the provider has no value"() {
```

Comment:
💭 I'd remove this test since we already have the one in `DefaultValueSourceProviderFactoryTest`. I can do that after the merge